test/Waives.Pipelines.Tests/DocumentProcessorFacts.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NSubstitute;
+using Waives.Pipelines.HttpAdapters;
+using Xunit;
+
+namespace Waives.Pipelines.Tests
+{
+    public class DocumentProcessorFacts
+    {
+        private readonly Func<Document, Task<WaivesDocument>> _docCreator;
+        private readonly Action<Exception, Document> _onDocumentException;
+        private readonly TestDocument _testDocument;
+
+        public DocumentProcessorFacts()
+        {
+            _docCreator = document =>
+            {
+                var waivesDocument = new WaivesDocument(document, Substitute.For<IHttpDocument>());
+                return Task.FromResult(waivesDocument);
+            };
+            _onDocumentException = (exception, document) => { };
+            _testDocument = new TestDocument(Generate.Bytes());
+        }
+
+        [Fact]
+        public async Task Create_WaivesDocument_from_Document()
+        {
+            Document capturedDoc = null;
+            var docActions = new List<Func<WaivesDocument, Task<WaivesDocument>>>
+            {
+                waivesDoc => {
+                    capturedDoc = waivesDoc.Source;
+                    return Task.FromResult(waivesDoc);
+                }
+            };
+            var sut = new DocumentProcessor(
+                _docCreator,
+                docActions,
+                _onDocumentException);
+
+            await sut.RunAsync(_testDocument);
+
+            Assert.Same(_testDocument, capturedDoc);
+        }
+
+        [Fact]
+        public async Task Runs_all_provided_actions_on_docs()
+        {
+            var fakeDocActions = FakeDocAction.AListOfDocActions(10);
```

Comment:
👌 Only need 3 items here :man_shrugging: Additionally could be a little neater with a `using static FakeDocAction`.
```suggestion
            var fakeDocActions = AListOfDocActions(3);
```